src/components/popup/details/details.jsx

```diff
@@ -45,30 +52,27 @@ export default class Details extends Component {
 	};
 
 	handleShowVendors = (filter) => {
-		this.state.vendors = this.getVendors(filter);
+		this.props.store.subsection = SECTION_VENDORS;
 		this.setState({
-			selectedPanelIndex: SECTION_VENDORS
+			vendors: this.filterVendors(filter)
 		});
 	};
 
 	handleBack = () => {
-		const { onCancel } = this.props;
-		const { selectedPanelIndex } = this.state;
-		this.setState({
-			selectedPanelIndex: Math.max(0, selectedPanelIndex - 1)
-		});
-		if (selectedPanelIndex === SECTION_PURPOSES) {
+		const { onCancel, store } = this.props;
+		const { subsection } = store;
+		store.subsection = Math.max(0, subsection - 1);
+		this.forceUpdate();
```

Comment:
Here we can use `store.updateSection(SECTION_DETAILS, Math.max(0, subsection - 1));`